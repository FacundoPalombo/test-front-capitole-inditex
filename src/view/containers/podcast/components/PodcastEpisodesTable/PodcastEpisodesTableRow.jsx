import React, { useMemo } from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.module.scss';

const PodcastEpisodesTableRow = ({
  trackId,
  trackName,
  releaseDate,
  trackTimeMillis,
}) => {
  const { trackMinutes, trackSeconds } = useMemo(
    () => ({
      trackMinutes: (trackTimeMillis - (trackTimeMillis % 60000)) / 60000,
      trackSeconds: ((trackTimeMillis % 60000) / 1000).toFixed(0),
    }),
    [trackTimeMillis]
  );

  const trackDateCreation = useMemo(
    () => new Date(releaseDate).toLocaleDateString(),
    [trackDateCreation]
  );

  return (
    <tr className={styles.table__row} key={trackId}>
      <th className={styles.table__row_item} align="left" headers="col-title">
        <Link to={`episode/${trackId}`} className={styles.table__row_link}>
          {trackName}
        </Link>
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-date"
      >
        {trackDateCreation}
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-duration"
      >
        {trackMinutes}
        min {trackSeconds}s
      </th>
    </tr>
  );
};

export default PodcastEpisodesTableRow;
