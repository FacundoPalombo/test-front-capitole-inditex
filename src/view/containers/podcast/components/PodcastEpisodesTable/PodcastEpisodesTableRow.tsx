import React, { useMemo } from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.module.scss';

const PodcastEpisodesTableRow = ({
  trackId,
  trackName,
  releaseDate,
  trackTimeMillis,
}) => {
  const trackMinutes = useMemo(
    () => (trackTimeMillis - (trackTimeMillis % 60000)) / 60000,
    [trackTimeMillis]
  );
  const trackSeconds = useMemo(
    () => ((trackTimeMillis % 60000) / 1000).toFixed(0),
    [trackTimeMillis]
  );
  const trackDateCreation = useMemo(
    () => new Date(releaseDate).toLocaleDateString(),
    [releaseDate]
  );

  return (
    <tr className={styles.table__row} key={trackId}>
      <th className={styles.table__row_item} align="left" headers="col-title">
        <Link to={`episode/${trackId}`} className={styles.table__row_link}>
          {trackName}
        </Link>
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-date"
      >
        {trackDateCreation}
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-duration"
      >
        {trackMinutes}
        min {trackSeconds}s
      </th>
    </tr>
  );
};

export default PodcastEpisodesTableRow;
