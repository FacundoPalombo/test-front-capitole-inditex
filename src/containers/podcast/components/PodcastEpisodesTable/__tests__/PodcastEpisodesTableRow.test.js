import React from 'react';
import { render } from '@testing-library/react';
import PodcastEpisodesTableRow from '../PodcastEpisodesTableRow';

jest.mock('react-router-dom', () => ({
  Link: ({ to, children }) => <a href={to}>{children}</a>,
}));

describe('PodcastEpisodesTableRow unit tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<PodcastEpisodesTableRow />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should build correct compoent with correct links and times', () => {
    const { getByRole } = render(
      <PodcastEpisodesTableRow
        trackId={12324}
        trackName="A la santa catalina chibirin chibirin bombom"
        releaseDate="2023-06-24T00:00:00-07:00"
        trackTimeMillis={3600 * 1000 + 2000}
      />
    );
    const link = getByRole('link', {
      name: 'A la santa catalina chibirin chibirin bombom',
    });

    expect(link).toHaveAttribute('href', 'episode/12324');

    const tableItemReleaseDate = getByRole('row', {
      name: 'A la santa catalina chibirin chibirin bombom 24/6/2023 60min 2s',
    });
    expect(tableItemReleaseDate).toBeInTheDocument();
  });
});
