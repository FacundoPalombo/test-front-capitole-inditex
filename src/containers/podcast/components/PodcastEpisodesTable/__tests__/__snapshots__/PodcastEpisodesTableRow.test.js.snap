// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastEpisodesTableRow unit tests should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="episode/undefined"
  />
  Invalid DateNaNmin NaNs
</DocumentFragment>
`;
